iocore/net/ALPNSupport.cc

```diff
@@ -0,0 +1,82 @@
+/** @file
+
+  ALPNSupport.cc provides implmentations for ALPNSupport methods
+
+  @section license License
+
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+
+#include "P_ALPNSupport.h"
+#include "P_SSLNextProtocolSet.h"
+#include "records/I_RecHttp.h"
+
+void
+ALPNSupport::clear()
```

Comment:
Hmm, I can live with this because NetVC already has `clear`, but I prefer to use its destructor if it's possible.

Reply:
Will the destructor get called for a free list allocated object?

Reply:
Not automatically, but we should call it explicitly. Otherwise destructors for member objects won't be called neither. Of course we should call constructors too for free list allocated objects too, many places don't do it though.